### ISSUE TITLE
Bug in Getregion

### ISSUE THREAD

**Issue by AceofCrafters** [mathuin/TopoMC#30]
When inputing (Windows 7):

python ./GetRegion.py --name PurcellLake --ymax 34.99837 --xmin -97.406287 --ymin 34.971437 --xmax -97.367234 --scale 1 --vscale 1 --trim 0 --sealevel 34 --disable-ore

I Get:

Creating new region PurcellLake...
Retrieving files...
Requested URLs for layer ID L1L...
Requested URLs for layer ID N3F...
Using cached file for layerID L1L
unzip:  cannot find either C:/Users/Brandon or C:/Users/Brandon.zip.
unzip:  cannot find either C:/Users/Brandon or C:/Users/Brandon.zip.
0...10...20...30.ERROR 4: `Skinner\Documents\GitHub\TopoMC\Regions\PurcellLake\D
atasets\L1L.vrt' does not exist in the file system,
and is not recognised as a supported dataset name.

Warning 1: Can't open Skinner\Documents\GitHub\TopoMC\Regions\PurcellLake\Datase
ts\L1L.vrt. Skipping it
..40...50...60..ERROR 4: `C:\Users\Brandon' does not exist in the file system,
and is not recognised as a supported dataset name.

Warning 1: Can't open C:\Users\Brandon. Skipping it
.70...80...90...100 - done.
ERROR 4: `Skinner\Documents\GitHub\TopoMC\Downloads\L1L\NLCD2001_LC_N33W096_v2.t
if' does not exist in the file system,
and is not recognised as a supported dataset name.

Warning 1: Can't open Skinner\Documents\GitHub\TopoMC\Downloads\L1L\NLCD2001_LC_
N33W096_v2.tif. Skipping it
ERROR 4: `C:\Users\Brandon' does not exist in the file system,
and is not recognised as a supported dataset name.

This is probably due to the fact that the location is:
C:\Users\Brandon Skinner\Documents\GitHub\TopoMC
The Folder: Brandon Skinner
Having a space in the name.

**Comment by mathuin** [comment]
I believe you are correct.  The lame workaround (work in a directory with a space-free path) is lame, so I will look at fixing this.  I have a Windows machine for testing, so this shouldn't take too long -- if you tinker with it and make a patch, send me a pull request!

Thanks again for the report!

Jack.

**Comment by mathuin** [comment]
Okay, I've checked in a fix to this.  When I run the code, the files are extracted properly, but I am getting SRS errors.  I don't think they're related.  Can you update your master checkout and try again?

I am in the middle of a minor reorganization of the code, which I will be pushing out next week in preparation for making a package.  I worry that I might have cherry picked the wrong patches. :-)

Let me know if it works for you!

Jack.

**Comment by mathuin** [comment]
Yeah, I broke master by a faulty cherry pick.  It's too late tonight to fix it -- develop works, though, and the command line will be the same for you since Windows is case insensitive. :-(

I will fix it this week, honest!

Jack.

**Comment by mathuin** [comment]
Okay, the bug has been fixed in the master branch.

I am now testing the entire set of commands on your region to ensure that it will work properly.

I expect the answer to be "yes", but now I want to be sure. :-)

**Comment by AceofCrafters** [comment]
Ill go ahead and test it now.
Thanks!
Brandon.

**Comment by jmhbrown** [comment]
Hey, I'm still getting a similar error. (I checked, I'm running the new-and-improved region.py) 

Here's my output:

[Jenny@jenny TopoMC]$ ./GetRegion.py --name Provincetown --ymax 42.0901 --xmin -70.2611 --ymin 42.0091 --xmax -70.1100
Creating new region Provincetown...
Retrieving files...
Requested URLs for layer ID L1L...
Requested URLs for layer ID N3F...
Downloading file from server for layerID L1L
100% |#########################################################################################################| Time: 00:00:03   4.01 M/s
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N42W069_v2.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N42W069_v2.tif  
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N42W069_v2.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N42W069_v2.tfw  
Downloading file from server for layerID L1L
100% |#########################################################################################################| Time: 00:00:00   2.73 M/s
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N39W069_v2.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N39W069_v2.tif  
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N39W069_v2.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N39W069_v2.tfw  
0...10...20...30...40...50...60...70...80...90...100 - done.
ERROR 4: `/home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N42W069_v2.tif /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N39W069_v2.tif' does not exist in the file system,
and is not recognised as a supported dataset name.

Warning 1: Can't open /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N42W069_v2.tif /home/Jenny/.minecraft/TopoMC/Downloads/L1L/NLCD2001_LC_N39W069_v2.tif. Skipping it
ERROR 4: `/home/Jenny/.minecraft/TopoMC/Regions/Provincetown/Datasets/L1L.vrt' does not exist in the file system,
and is not recognised as a supported dataset name.

Downloading file from server for layerID N3F
100% |#########################################################################################################| Time: 00:00:38   2.47 M/s
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n43w071.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n43w071/floatn43w071_13.flt  
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n43w071.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n43w071/floatn43w071_13.hdr  
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n43w071.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n43w071/floatn43w071_13.prj  
Downloading file from server for layerID N3F
100% |#########################################################################################################| Time: 00:00:52   2.16 M/s
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n42w071.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n42w071/floatn42w071_13.flt  
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n42w071.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n42w071/floatn42w071_13.hdr  
Archive:  /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n42w071.zip
  inflating: /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n42w071/floatn42w071_13.prj  
0...10...20...30...40...50...60...70...80...90...100 - done.
ERROR 4: `/home/Jenny/.minecraft/TopoMC/Downloads/N3F/n43w071/floatn43w071_13.flt /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n42w071/floatn42w071_13.flt' does not exist in the file system,
and is not recognised as a supported dataset name.

Warning 1: Can't open /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n43w071/floatn43w071_13.flt /home/Jenny/.minecraft/TopoMC/Downloads/N3F/n42w071/floatn42w071_13.flt. Skipping it
ERROR 4: `/home/Jenny/.minecraft/TopoMC/Regions/Provincetown/Datasets/N3F.vrt' does not exist in the file system,
and is not recognised as a supported dataset name.

It looks like GetRegion.py or region.py is concatenating some file paths oddly? I'll post an update if I get it working.

**Comment by mathuin** [comment]
This is what I get for trying to commit a "quick fix" during finals! :-)

Short form:  I fixed it but forgot to push the fix, then broke it when I tried to merge it.

If you update your checkout _now_ and try again, it should work as you expect.  The odd concatenation is due to trying to concatenate files that could have spaces in their name -- my first attempt was "one two three four" when I needed '"one two" "three four"'.  If that makes any sense.  It does to me, at least.

Some day when the GDAL Python API is extended to cover warping, this will all be moot, promise.  Alternatively, if the mapserver idea works, maybe all the warping could be done on that side.  Hmm.  Anyway.

I know it works because I just ran the test myself before telling you here.  Let me know if it works for you?

Jack.
